src/prerender.js

```diff
@@ -31,7 +31,7 @@ export default async function prerender(vnode, options) {
 	const render = () => {
 		if (++tries > maxDepth) return;
 		try {
-			return renderToString(vnode);
+			return renderToStringAsync(vnode);
 		} catch (e) {
 			if (e && e.then) return e.then(render);
 			throw e;
```

Comment:
_I think_ this whole block can be replaced? `renderToStringAsync` should handle that itself.

Reply:
Is this your assumption?

```js
const render = () => {
	if (++tries > maxDepth) return;
	return renderToStringAsync(vnode);
};
```

Reply:
Can get rid of `render` altogether -- `renderToStringAsync` tracks depth and whatnot itself:

https://github.com/preactjs/preact-render-to-string/blob/e3326e2ec92e52032d8380fdc6ca5aed70fbcc48/src/index.js#L115-L128

Reply:
I see, thanks.
Fixed here
https://github.com/preactjs/preact-iso/pull/10/commits/bb3ddb0210d8f524c1e165aaf08cfc238a5a22b4